### ISSUE TITLE
RDS replica creation requires double apply

### ISSUE THREAD

**Issue by bethge** [terraform-providers/terraform-provider-aws#2909]
### Terraform Version
```
Terraform v0.11.1
+ provider.aws v1.6.0
```

### Affected Resource(s)
- `aws_db_instance`

### Terraform Configuration Files
```hcl
provider "aws" {
  region = "us-east-1"
}

resource "aws_vpc" "main" {
  cidr_block = "10.0.0.0/16"
}

resource "aws_subnet" "main-a" {
  vpc_id            = "${aws_vpc.main.id}"
  availability_zone = "us-east-1a"
  cidr_block        = "10.0.1.0/24"
}

resource "aws_subnet" "main-b" {
  vpc_id            = "${aws_vpc.main.id}"
  availability_zone = "us-east-1b"
  cidr_block        = "10.0.2.0/24"
}

resource "aws_db_subnet_group" "main" {
  subnet_ids = ["${aws_subnet.main-a.id}", "${aws_subnet.main-b.id}"]
}

resource "aws_db_instance" "master" {
  allocated_storage       = 10
  engine                  = "mysql"
  instance_class          = "db.t2.micro"
  username                = "foo"
  password                = "barbar12"
  backup_retention_period = 1
  apply_immediately       = true
  db_subnet_group_name    = "${aws_db_subnet_group.main.name}"
}

resource "aws_db_instance" "replica" {
  replicate_source_db = "${aws_db_instance.master.identifier}"
  allocated_storage   = 15
  instance_class      = "db.t2.micro"
}

```

### Expected Behavior
There are no changes to be applied after an execution of `terraform apply`.

### Actual Behavior
There are changes to be applied:
```
  ~ aws_db_instance.replica
      allocated_storage: "10" => "15"
```

### Steps to Reproduce

1. `terraform apply`
2. `terraform plan`

### Important Factoids
- The [`CreateDBInstanceReadReplica` action][1] only allows for a subset of parameters to be applied on creation.

[1]: https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_CreateDBInstanceReadReplica.html

Iiuc the limitations of the `CreateDBInstanceReadReplica` AWS API call is the issue. Certain parameters can not be configured during DB creation but must be changed after creation. Is there some way for terraform to circumvent the limitations of the AWS API in this situation? E.g. by creating and then updating the `db_instance` in one apply execution.

I would be more than willing to work on a PR to resolve this issue, assuming it is within the realm of fixability.

**Comment by Bryji** [comment]
I'm having a similar issue (as linked above in the terraform-aws-rds repo), read replicas in a multi-region MySQL set up have multi-az set to false. It looks like it remains false after re-applying config also (but on one occasion the change did take after a re-apply).

**Comment by bflad** [comment]
This has been released in [version 1.34.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1340-august-30-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
